iothub/service/tests/DeserializationTests.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class DeserializationTests
+    {
+        [TestMethod]
+        public void Twin_JsonDateParse_Ok()
+        {
+            const string ExpectedDeviceId = "test";
+            const string JsonString = @"
+{
```

Comment:
nit: the indentation here seems off. Does it look okay in visual studio? I've had some discrepancies between VS and github before

Reply:
This often happens with json strings. We don't want formatting spacing inside the string because it is multiline.

Reply:
The alternative is to use + signs on the end of each line. It gets kind of ugly that way too. There's no solution I love, unless you create an anonymous object and then json serialize it to produce a string.